packages/bvaughn-architecture-demo/components/sources/CurrentLineHighlight.tsx

```diff
@@ -31,12 +32,22 @@ function CurrentLineHighlightSuspends({ lineNumber, sourceId }: Props) {
   if (pauseId !== null && frameId !== null) {
     const pauseData = getPauseDataSuspense(client, pauseId);
     if (pauseData !== null && pauseData.frames) {
+      const correspondingSourceIds = client.getCorrespondingSourceIds(sourceId);
       const selectedFrame = pauseData.frames.find(frame => frame.frameId === frameId);
       if (selectedFrame) {
         if (
-          selectedFrame.location.find(
-            location => location.line === lineNumber && location.sourceId === sourceId
-          )
+          selectedFrame.location.find(location => {
+            if (correspondingSourceIds.includes(location.sourceId)) {
+              const correspondingLocations = client.getCorrespondingLocations(location);
+              return (
+                correspondingLocations.findIndex(
+                  correspondingLocation =>
+                    correspondingLocation.line === lineNumber &&
+                    correspondingLocation.sourceId === sourceId
```

Comment:
My head hurts

Reply:
Mine too.

Reply:
The logic here is basically saying:

This is the file that's open.

If the currently selected frame matches this line– for _any_ source that "corresponds to" this one, show the highlight.